airflow/jobs/scheduler_job_runner.py

```diff
@@ -819,19 +821,20 @@ def _execute(self) -> int | None:
             )
 
         try:
-            self.job.executor.job_id = self.job.id
-            if self.processor_agent:
-                self.log.debug("Using PipeCallbackSink as callback sink.")
-                self.job.executor.callback_sink = PipeCallbackSink(
-                    get_sink_pipe=self.processor_agent.get_callbacks_pipe
-                )
-            else:
-                from airflow.callbacks.database_callback_sink import DatabaseCallbackSink
+            for executor in self.job.executors:
+                executor.job_id = self.job.id
+                if self.processor_agent:
+                    self.log.debug("Using PipeCallbackSink as callback sink.")
```

Comment:
Since it is not possible for executors to have different callback sinks, it’s probably better to only show this mesage once. The if-else check should only be done once to begin with.

Reply:
> The if-else check should only be done once to begin with.

Fair point, I can do that check just once before the loop

> Since it is not possible for executors to have different callback sinks

Do you mean to say that executors can't have different _types_ of callback sinks? Or that they should literally share the same in-memory sink object between all executors?

@uranusjr

Reply:
any reason they should not use the same sink instance?

Reply:
Happy to share the sink between all executors, I'll make the change along with the above change :)

Reply:
Changes made and PR updated!